core/trino-main/src/main/java/io/trino/execution/buffer/OutputBufferStateMachine.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution.buffer;
+
+import io.trino.execution.StateMachine;
+import io.trino.execution.TaskId;
+
+import java.util.concurrent.Executor;
+
+import static io.trino.execution.buffer.BufferState.FAILED;
+import static io.trino.execution.buffer.BufferState.FINISHED;
+import static io.trino.execution.buffer.BufferState.FLUSHING;
+import static io.trino.execution.buffer.BufferState.NO_MORE_BUFFERS;
+import static io.trino.execution.buffer.BufferState.NO_MORE_PAGES;
+import static io.trino.execution.buffer.BufferState.OPEN;
+import static io.trino.execution.buffer.BufferState.TERMINAL_BUFFER_STATES;
+
+public class OutputBufferStateMachine
```

Comment:
It looks like we are not using `boolean stateChanged` return values most of the time. Would that make sense to return `void` for methods where we do not care about returned value.

Reply:
We need a boolean for most of the methods. The return value is not used for `noMoreBuffers` and `fail`. But I thought it might be better to be consistent with other methods.